tests/test_parametric_reactors/test_SubmersionTokamak.py

```diff
@@ -0,0 +1,273 @@
+            "tf_pf_SubmersionTokamak/outboard_firstwall.stp",
+            "tf_pf_SubmersionTokamak/supports.stp",
+            "tf_pf_SubmersionTokamak/blanket.stp",
+            "tf_pf_SubmersionTokamak/outboard_rear_blanket_wall.stp",
+            "tf_pf_SubmersionTokamak/Graveyard.stp",
+            "tf_pf_SubmersionTokamak/pf_coil_cases.stp"
+        ]
+
+        for output_filename in output_filenames:
+            assert Path(output_filename).exists() is True
+        os.system("rm -r tf_pf_SubmersionTokamak")
+
+    def test_rotation_angle_warning(self):
+        """Creates a SubmersionTokamak with rotation_angle = 360 and checks that the
+        correct warning message is printed."""
+
+        def warning_trigger():
+            try:
+                self.test_reactor.rotation_angle = 360
+                self.test_reactor._rotation_angle_check()
+            except BaseException:
+                pass
+        msg = "360 degree rotation may result in a " + \
+            "Standard_ConstructionError or AttributeError"
+        with pytest.warns(UserWarning, match=msg):
+            warning_trigger()
+
+    def test_submersion_reactor_hash_value(self):
+        """Creates a submersion reactor and checks that all shapes in the reactor are created
+        when .shapes_and_components is first called. Checks that when .shapes_and_components
+        is called again with no changes to the reactor, the shapes in the reactor are
+        reconstructed and the previously constructed shapes are returned. Checks that when
+        .shapes_and_components is called again with no changes to the reactor, the shapes in
+        the reactor are reconstructed and these new shapes are returned. Checks that the
+        reactor_hash_value is only updated when the reactor is reconstructed."""
+
+        self.test_reactor.pf_coil_radial_thicknesses = [30, 30, 30, 30]
+        self.test_reactor.pf_coil_vertical_thicknesses = [30, 30, 30, 30]
+        self.test_reactor.pf_coil_to_tf_coil_radial_gap = 50
+        self.test_reactor.pf_coil_case_thickness = 10
+        self.test_reactor.outboard_tf_coil_radial_thickness = 30
+        self.test_reactor.outboard_tf_coil_poloidal_thickness = 30
+        self.test_reactor.tf_coil_to_rear_blanket_radial_gap = 20
+        self.test_reactor.number_of_tf_coils = 16
+
+        assert self.test_reactor.reactor_hash_value is None
+        for key in [
+            "_inboard_tf_coils",
+            "_center_column_shield",
+            "_plasma",
+            "_inboard_firstwall",
+            "_inboard_blanket",
+            "_firstwall",
+            "_divertor",
+            "_blanket",
+            "_supports",
+            "_outboard_rear_blanket_wall_upper",
+            "_outboard_rear_blanket_wall_lower",
+            "_outboard_rear_blanket_wall",
+            "_tf_coil",
+            "_pf_coil",
+            "_pf_coils_casing"
+        ]:
+            assert key not in self.test_reactor.__dict__.keys()
```

Comment:
**SUGGESTION**

dicts directly support membership tests, so this could be: 
```py
key not in self.test_reactor.__dict__
```

Reply:
fixed in commit 3ef64cf4